#596 Declare supported operations (#652)
thin-edge/thin-edge.io: docs/src/tutorials/supported_operations.md

# thin-edge.io Supported Operations

## Supported Operations concepts

### Device operations

IoT devices often do more than just send data to the cloud. They also do things like:

* receive triggers from the operator
* reboot on demand
* install or remove software

> These are all operations that are supported by Cumulocity IoT and other cloud providers.
> In thin-edge.io we support these operations by writing a plugin that implements the
> [`Operation`](https://cumulocity.com/guides/reference/operation) interface and registering it with the
> thin-edge.io Supported Operations API.

Reviewer: This introduction can be improved. Notably, it seems better to link the Cumulocity operation doc to Cumulocity  rather than to thin-edge. By the way, the link is broken.

```suggestion
These operations that are supported by [Cumulocity IoT](https://cumulocity.com/api/10.11.0/#section/Device-management-library) and other cloud providers. On thin-edge.io the support for one such operation can be added using the thin-edge.io Supported Operations API.
```

### thin-edge.io Supported Operations API

> The Supported Operations utilises file system to add and remove operations. Special file placed in `/etc/tedge/operations` directory will indicate that the operation is supported.

Reviewer: ```suggestion
The Supported Operations utilises the file system to add and remove operations. A special file placed in `/etc/tedge/operations` directory will indicate that an operation is supported.
```

The specification for the operation files is described in thin-edge.io specifications repository[src/supported-operations/README.md](https://github.com/thin-edge/thin-edge.io-specs/blob/a99a8cbf78a4c4c9637fb1794797cb2fb468a0f4/src/supported-operations/README.md)

## thin-edge.io List of Supported Operations

> thin-edge.io comes with some operations that are supported `natively` by thin-edge.io.
> Following operations are supported:

Reviewer: ```suggestion
thin-edge.io supports natively the following operations:
```

* Software Update
* Software Update Log Upload
* Restart

The list is growing as we support more operations, but is not exhaustive and we encourage you to contribute to the list.

## How to use Supported Operations

### Listing current operations

> You can obtain current list of supported operations by running listing content of `/etc/tedge/operations` directory.

Reviewer: ```suggestion
You can obtain the current list of supported operations by listing the content of the `/etc/tedge/operations` directory.
```
